src/main/java/frc/robot/VisionProcessor.java

```diff
@@ -132,19 +145,24 @@ public void process(Mat sourceFrame) {
         ArrayList<RotatedRect> initial = minimumBoundingRectangle(parent.filterContoursOutput());
         ArrayList<RotatedRect> findContours = minimumBoundingRectangle(parent.findContoursOutput());
         ArrayList<RotatedRect> ok = new GibberishRectangleFilter().filter(initial);
-        ArrayList<RotatedRect> selected = new DumbAndAloneRectangleFilter().filter(ok);
-
+        index = new CenterRectangleIndexFinder().getIndex(ok);
+        if(lockTracker.isTargetLockOn()){
+            lockTracker.checkLock(ok.size());
+        }else{
+            lockTracker.setupLock(ok.size(), index);
+        }        
+        ArrayList<RotatedRect> selected = new FinalVisionTargetFilter().filter(ok, index);
 
         timer.start(TIMERS.OUTPUT);
-        drawRectanglesOnImage(resizedImage, initial, COLORS.BLUE);
-        drawRectanglesOnImage(resizedImage, findContours, COLORS.PURPLE);
-        drawRectanglesOnImage(resizedImage, ok, COLORS.RED);
-        drawRectanglesOnImage(resizedImage, selected, COLORS.YELLOW);
+        drawRectanglesOnImage(resizedImage, initial, COLORS.BLUE, false);
+        drawRectanglesOnImage(resizedImage, findContours, COLORS.PURPLE, false);
+        drawRectanglesOnImage(resizedImage, ok, COLORS.RED, false);
+        drawRectanglesOnImage(resizedImage, selected, COLORS.BLUE, targetLock);
```

Comment:
are you sure you want to display selected ones in the same color as the initial ones?